### PR TITLE
[NETBEANS-4340] Added PHP Mess Detector options configuration

### PR DESCRIPTION
**Pull request by KacerCZ**
https://issues.apache.org/jira/browse/NETBEANS-4340

Added CLI options to PHP Mess Detector configuration.

Before:
options-before
customizer-before

After:
options-after
customizer-after

### CODE REVIEW

**Comment by KacerCZ** [comment]
@junichi11 @tmysik I found issue with customizer panel (Source > Inspect > Configuration > Manage) that dialog is too small and Options text field is not visible. You have to resize the dialog manually to see it.

How can this be fixed?
Is it issue with this panel or in including [AdjustConfigurationPanel](https://github.com/apache/netbeans/blob/master/ide/code.analysis/src/org/netbeans/modules/analysis/ui/AdjustConfigurationPanel.java)?

**Comment by tmysik** [comment]
Not sure if it is applicable also here but usually a minimum size of the dialog can be defined.

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/MessDetectorAnalyzerImpl.java [hunk 1]
```diff
@@ -223,6 +224,20 @@ private FileObject getValidRuleSetFile() {
...
+    private String getValidOptions() {
+        String options = null;
+        Preferences settings = context.getSettings();
+        if (settings != null) {
+            options = settings.get(MessDetectorCustomizerPanel.OPTIONS, null);
+        }
+        if (options == null) {
+            options = AnalysisOptions.getInstance().getMessDetectorOptions();
+        }
+        assert options != null;
```
Comment (junichi11) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/MessDetectorAnalyzerImpl.java:237]: Remove?

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/AnalysisOptions.java [hunk 2]
```diff
@@ -142,6 +143,15 @@ public void setMessDetectorRuleSetFilePath(String ruleSetFilePath) {
         getPreferences().put(MESS_DETECTOR_RULE_SET_FILE, ruleSetFilePath);
     }
 
+    @CheckForNull
+    public String getMessDetectorOptions() {
+        return getPreferences().get(MESS_DETECTOR_OPTIONS, "");
```
Comment (junichi11) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/AnalysisOptions.java:148]: `null` or add `NOI18N`

**Comment by junichi11** [comment]
> I found issue with customizer panel (Source > Inspect > Configuration > Manage) that dialog is too small and Options text field is not visible. You have to resize the dialog manually to see it.
> 
> How can this be fixed?
> Is it issue with this panel or in including [AdjustConfigurationPanel](https://github.com/apache/netbeans/blob/master/ide/code.analysis/src/org/netbeans/modules/analysis/ui/AdjustConfigurationPanel.java)?

We should separate it from this issue.

@jlahoda Is `AdjustConfigurationPanel` resizable automatically?

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/MessDetectorAnalyzerImpl.java (hunk 1 above)
Comment (KacerCZ) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/MessDetectorAnalyzerImpl.java]: I removed the assert.

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/AnalysisOptions.java (hunk 2 above)
Comment (KacerCZ) [reply php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/AnalysisOptions.java]: I changed default value to `null`.

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/analyzer/MessDetectorCustomizerPanel.java [hunk 3]
```diff
@@ -86,28 +87,30 @@ public MessDetectorCustomizerPanel(Analyzer.CustomizerContext<Void, MessDetector
     }
 
     private void init() {
+        setEnabledCheckBox();
+        setRuleSetsList();
+        setRuleSetFileTextField();
+        setOptionsTextField();
```
Comment (tmysik) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/analyzer/MessDetectorCustomizerPanel.java:93]: Perhaps a bit better name for these methods would be `init...`.

**Comment by tmysik** [comment]
@junichi11 @KacerCZ 

I am going to merge it soon - OK?

**Comment by junichi11** [comment]
@tmysik It's OK with me :)

**Comment by KacerCZ** [comment]
Thank you.

**Pull request merged**